cvat/apps/iam/serializers.py

```diff
@@ -27,6 +32,44 @@ def get_cleaned_data(self):
 
         return data
 
+    def validate_email(self, email):
+        email = get_adapter().clean_email(email)
+        if app_settings.UNIQUE_EMAIL:
+            if email and email_address_exists(email):
+                user = get_dummy_user(email)
+                if not user:
+                    raise serializers.ValidationError(
+                        ('A user is already registered with this e-mail address.'),
+                    )
+
+        return email
+
+    def save(self, request):
+        adapter = get_adapter()
+        user = adapter.new_user(request)
```

Comment:
Wouldn't it make more sense to do this after you verified that no dummy user exists?

Reply:
Yes, in that case we dont need to call new_user. Fixed